packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterTest.kt

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package com.facebook.react.uimanager
+
+import android.view.View
+import com.facebook.react.bridge.JSApplicationIllegalArgumentException
+import com.facebook.react.bridge.JavaOnlyArray
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.uimanager.annotations.ReactPropGroup
+import org.assertj.core.api.Assertions
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+/** Test updating view through [ViewManager] with [ReactProp] and [ReactPropGroup] annotations. */
+@RunWith(RobolectricTestRunner::class)
+class ReactPropAnnotationSetterTest {
+  interface ViewManagerUpdatesReceiver {
+    fun onBooleanSetterCalled(value: Boolean)
+
+    fun onIntSetterCalled(value: Int)
+
+    fun onDoubleSetterCalled(value: Double)
+
+    fun onFloatSetterCalled(value: Float)
+
+    fun onStringSetterCalled(value: String?)
+
+    fun onBoxedBooleanSetterCalled(value: Boolean?)
+
+    fun onBoxedIntSetterCalled(value: Int?)
+
+    fun onArraySetterCalled(value: ReadableArray?)
+
+    fun onMapSetterCalled(value: ReadableMap?)
+
+    fun onFloatGroupPropSetterCalled(index: Int, value: Float)
+
+    fun onIntGroupPropSetterCalled(index: Int, value: Int)
+
+    fun onBoxedIntGroupPropSetterCalled(index: Int, value: Int?)
+  }
+
+  @Suppress("UNUSED_PARAMETER")
+  private inner class ViewManagerUnderTest
+  constructor(val mViewManagerUpdatesReceiver: ViewManagerUpdatesReceiver) :
+    ViewManager<View?, ReactShadowNode<*>?>() {
+    override fun getName(): String {
+      return "RedpandasLivestreamVideoView"
+    }
+
+    override fun createShadowNodeInstance(): ReactShadowNode<*>? {
+      Assertions.fail<Any>("This method should not be executed as a part of this test")
+      return null
+    }
+
+    override fun getShadowNodeClass(): Class<out ReactShadowNode<*>> {
+      return ReactShadowNode::class.java
+    }
+
+    override fun createViewInstance(reactContext: ThemedReactContext): View {
+      Assertions.fail<Any>("This method should not be executed as a part of this test")
+      return View(RuntimeEnvironment.getApplication())
+    }
```

Comment:
```suggestion
    override fun createViewInstance(reactContext: ThemedReactContext): View =
      error("This method should not be executed as a part of this test")
```

Please apply the same change to all the similar methods that we don't expect to call inside the tests